### PR TITLE
Issue GL: 132 BuddyPress activity upload issue with Microsoft Edge

### PR DESCRIPTION
**Pull request by yahilmadakiya**

### CODE REVIEW

File: app/assets/js/rtMedia.js [hunk 1]
```diff
@@ -674,11 +674,12 @@ jQuery( 'document' ).ready( function( $ ) {
 
 	// Arrange media into masonry view right after upload or clicking on readmore link to activity without pageload
 	jQuery( document ).ajaxComplete( function( event, xhr, settings ) {
-		var params = new URLSearchParams( settings.data );
-		if ( ( params.get('action') == 'post_update' || params.get('action') == 'get_single_activity_content' || params.get('action') == 'activity_get_older_updates' ) && typeof rtmedia_masonry_layout != 'undefined' && rtmedia_masonry_layout == 'true' && typeof rtmedia_masonry_layout_activity != 'undefined' && rtmedia_masonry_layout_activity == 'true' ) {
+		var get_action = get_parameter( 'action', settings.data );
+
+		if ( ( get_action == 'post_update' || get_action == 'get_single_activity_content' || get_action == 'activity_get_older_updates' ) && typeof rtmedia_masonry_layout != 'undefined' && rtmedia_masonry_layout == 'true' && typeof rtmedia_masonry_layout_activity != 'undefined' && rtmedia_masonry_layout_activity == 'true' ) {
```
Comment (milindmore22) [inline app/assets/js/rtMedia.js]: please add strict comparison with Yoda condition

File: app/assets/js/rtMedia.js [hunk 2]
```diff
@@ -1035,3 +1036,26 @@ function rtmedia_activity_masonry() {
 		rtm_masonry_reload( jQuery('#activity-stream .rtmedia-activity-container .rtmedia-list') );
 	}, 1000 );
 }
+
+/**
+ * Get specific parameter value from Query string.
+ */
+function get_parameter( parameter, data ) {
+	if ( ! parameter ) {
```
Comment (milindmore22) [inline app/assets/js/rtMedia.js]: add new line for readability

File: app/assets/js/rtMedia.js [hunk 3]
```diff
@@ -1035,3 +1036,26 @@ function rtmedia_activity_masonry() {
 		rtm_masonry_reload( jQuery('#activity-stream .rtmedia-activity-container .rtmedia-list') );
 	}, 1000 );
 }
+
+/**
+ * Get specific parameter value from Query string.
```
Comment (milindmore22) [inline app/assets/js/rtMedia.js]: please specify params and return types

**Comment by BhargavBhandari90** [comment]
@milindmore22 All changes are done. Please review.

**Comment by milindmore22** [comment]
@chandrapatel please do a final review

File: app/assets/js/rtMedia.backbone.js [hunk 4]
```diff
@@ -954,6 +954,17 @@ jQuery( document ).ready( function( $ ) {
...
 
+			/**
+			 * NOTE: Do not change.
+			 * ISSUE: BuddyPress activity upload issue with Microsoft Edge
+			 * GL: 132 [ http://git.rtcamp.com/rtmedia/rtMedia/issues/132 ]
+			 * Reason: Trigger event not working for hidden element in Microsoft Edge browser
+			 * Condition to check current browser.
+			 */
+			if ( /Edge/.test( navigator.userAgent ) ) {
+				jQuery( this ).closest( '.rtm-upload-button-wrapper' ).find( 'input[type=file]' ).click();
```
Comment (pradeep910) [inline app/assets/js/rtMedia.backbone.js]: @BhargavBhandari90 Looks like this code runs on document ready on Edge, regardless if its clicked or not. 
This code should run on Button click event. Please check. @chandrapatel @milindmore22 
`jQuery( this ).closest( '.rtm-upload-button-wrapper' ).find( 'input[type=file]' ).click();`

**Pull request merged**